lapack/testlapack/dtrexc.go

```diff
@@ -205,16 +199,126 @@ func testDtrexc(t *testing.T, impl Dtrexcer, compq lapack.UpdateSchurComp, tmat
 				continue
 			}
 			if q.Data[i*q.Stride+j]-qCopy.Data[i*qCopy.Stride+j] != 0 {
-				t.Errorf("%v: unexpected modification of Q[%v,%v]", prefix, i, j)
+				t.Errorf("%v: unexpected modification of Q[%v,%v]", name, i, j)
 			}
 		}
 	}
-	// Check that Qᵀ TOrig Q == T.
-	tq := eye(n, n)
-	blas64.Gemm(blas.NoTrans, blas.NoTrans, 1, tmatCopy, q, 0, tq)
-	qtq := eye(n, n)
-	blas64.Gemm(blas.Trans, blas.NoTrans, 1, q, tq, 0, qtq)
-	if !equalApproxGeneral(qtq, tmat, tol) {
-		t.Errorf("%v: Qᵀ (initial T) Q and (final T) are not equal", prefix)
+	// Check that Qᵀ * TOrig * Q == T
+	qt := zeros(n, n, n)
+	blas64.Gemm(blas.Trans, blas.NoTrans, 1, q, tmatCopy, 0, qt)
+	qtq := cloneGeneral(tmat1)
+	blas64.Gemm(blas.NoTrans, blas.NoTrans, -1, qt, q, 1, qtq)
+	resid = dlange(lapack.MaxColumnSum, n, n, qtq.Data, qtq.Stride)
+	if resid > float64(n)*tol {
+		t.Errorf("%v: mismatch between Qᵀ*(initial T)*Q and (final T); resid=%v, want<=%v",
+			name, resid, float64(n)*tol)
+	}
+}
+
+func residualOrthogonal(q blas64.General, rowwise bool) float64 {
+	m, n := q.Rows, q.Cols
+	if m == 0 || n == 0 {
+		return 0
+	}
+	var transq blas.Transpose
+	if m < n || (m == n && rowwise) {
+		transq = blas.NoTrans
+	} else {
+		transq = blas.Trans
+	}
+	minmn := min(m, n)
+
+	// Set work = I.
+	work := blas64.Symmetric{
+		Uplo:   blas.Upper,
+		N:      minmn,
+		Data:   make([]float64, minmn*minmn),
+		Stride: minmn,
+	}
+	for i := 0; i < minmn; i++ {
+		work.Data[i*work.Stride+i] = 1
+	}
+
+	// Compute
+	//  work = work - Q * Qᵀ = I - Q * Qᵀ
+	// or
+	//  work = work - Qᵀ * Q = I - Qᵀ * Q
+	blas64.Syrk(transq, -1, q, 1, work)
+	return dlansy(lapack.MaxColumnSum, blas.Upper, work.N, work.Data, work.Stride)
+}
+
+func dlansy(norm lapack.MatrixNorm, uplo blas.Uplo, n int, a []float64, lda int) float64 {
```

Comment:
As with `dlange` I'm not sure if it's better to add these methods to the tested interface or use these duplicated but local, independent implementations. I slightly prefer this but you may have a different opinion.

Reply:
I'm fine with this.